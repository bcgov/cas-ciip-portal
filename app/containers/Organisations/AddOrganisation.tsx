import React, { useState } from "react";
import { graphql, createFragmentContainer, RelayProp } from "react-relay";
import { Dropdown, FormControl, Card, Table } from "react-bootstrap";
import { AddOrganisation_query } from "AddOrganisation_query.graphql";
import AddOrganisationModal from "components/AddOrganisationModal";
import createOrganisationMutation from "mutations/organisation/createOrganisationMutation";
import Organisation from "./Organisation";

interface Props {
  query: AddOrganisation_query;
  relay: RelayProp;
  orgInput: string;
  selectedOrg: number;
  handleInputChange: (event: any) => void;
  handleOrgChange: (event: any) => void;
}

export const AddOrganisationComponent: React.FunctionComponent<Props> = (
  props
) => {
  const [selectedOrgDetails, setSelectedOrgDetails] = useState(null);
  const { allOrganisations } = props.query;
  const changeInput = (event) => {
    event.stopPropagation();
    event.preventDefault();
    event.persist();
    props.handleInputChange(event.target.value);
  };

  const selectOrg = (name, id, tradeName, cra, swrsReportId) => {
    props.handleOrgChange(id);
    props.handleInputChange("");
    setSelectedOrgDetails({ name, id, tradeName, cra, swrsReportId });
  };

  const handleAddOrganisation = async (variables) => {
    const { environment } = props.relay;
    await createOrganisationMutation(environment, variables);
  };

  return (
    <div>
      <Card style={{ marginTop: "50px" }}>
        <Card.Body>
          <Card.Title>Search for an operation: </Card.Title>
          <Dropdown className="search-dropdown">
            <Dropdown.Toggle id="org-dropdown" className="search-toggle">
              Find Operation
            </Dropdown.Toggle>
            <Dropdown.Menu>
              <FormControl
                placeholder="Search..."
                className="mx-3 my-2 w-auto"
                onChange={changeInput}
              />
              <div className="org-scroll">
                {allOrganisations.edges.map(({ node }) => {
                  return (
                    <Organisation
                      key={node.id}
                      select
                      organisation={node}
                      orgInput={props.orgInput}
                      selectOrg={selectOrg}
                    />
                  );
                })}
              </div>
            </Dropdown.Menu>
          </Dropdown>
          {props.selectedOrg !== null && (
            <Table
              style={{
                textAlign: "center",
                margin: "10px 0 2em 0",
                borderBottom: "1px solid rgb(0,0,0,0.1)",
              }}
            >
              <thead>
                <tr>
                  <th>Operator Name</th>
                  <th>Trade Name</th>
                  <th>CRA Business Number</th>
                  <th>SWRS Report Id</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>{selectedOrgDetails?.name}</td>
                  <td>{selectedOrgDetails?.tradeName}</td>
                  <td>{selectedOrgDetails?.cra}</td>
                  <td>{selectedOrgDetails?.swrsReportId}</td>
                </tr>
              </tbody>
            </Table>
          )}
          <AddOrganisationModal onAddOrganisation={handleAddOrganisation} />
          <style jsx>
            {`
              .org-scroll {
                max-height: 250px;
                overflow: hidden;
                overflow-y: scroll;
              }

              thead {
                background: #036;
                color: #fff;
              }
            `}
          </style>
        </Card.Body>
      </Card>
    </div>
  );
};

export default createFragmentContainer(AddOrganisationComponent, {
  query: graphql`
    fragment AddOrganisation_query on Query {
      allOrganisations {
        edges {
          node {
            id
            ...Organisation_organisation
          }
        }
      }
    }
  `,
});
