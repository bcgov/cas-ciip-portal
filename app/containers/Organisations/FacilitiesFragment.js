import React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { Card, Accordion } from 'react-bootstrap';

const FacilitiesFragment = (props) => {
    return (<Accordion.Collapse eventKey={props.facility.organisationId}><Card.Body data-testid="card" style={{ borderBottom: '1px solid #d3d3d3' }}>
        <div style={{ fontWeight: 'bold' }}>
            {props.facility.facilityName}

            <Button variant="primary" style={{ marginLeft: '30px' }}>
                Apply for CIIP for this facility
      </Button>
        </div>
        <p style={{ marginTop: '10px' }}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor.
    </p>
    </Card.Body></Accordion.Collapse>);
}

// export default FacilitiesFragment;
export default createFragmentContainer(FacilitiesFragment, {
    facility: graphql`
    fragment FacilitiesFragment_facility on Facility{
        id
        rowId
        organisationId
        facilityName
    }`
});