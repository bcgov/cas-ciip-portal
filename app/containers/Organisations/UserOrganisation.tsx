import React from 'react';
import {graphql, createFragmentContainer} from 'react-relay';
import {Badge} from 'react-bootstrap';
import Link from 'next/link';

export const UserOrganisationComponent = (props) => {
  const {userOrganisation} = props;
  if (!userOrganisation) {
    return null;
  }

  const statusBadgeColor = {
    REJECTED: 'danger',
    PENDING: 'info',
    APPROVED: 'success'
  };

  return (
    <tr>
      <td>{userOrganisation.organisationByOrganisationId.operatorName}</td>
      <td>
        <Badge
          pill
          style={{width: '100%', padding: '8px'}}
          variant={statusBadgeColor[userOrganisation.status]}
        >
          {userOrganisation.status}
        </Badge>
      </td>
      <td>
        {' '}
        {userOrganisation.status === 'APPROVED' ? (
          <Link
            href={{
              pathname: '/reporter/facilities',
              query: {
                relayVars: JSON.stringify({
                  organisationRowId:
                    userOrganisation.organisationByOrganisationId.rowId
                })
              }
            }}
          >
            <a className="btn btn-outline-primary">View Facilities</a>
          </Link>
        ) : null}
      </td>
      <style jsx>
        {`
          td {
            padding: 20px;
          }
        `}
      </style>
    </tr>
  );
};

export default createFragmentContainer(UserOrganisationComponent, {
  userOrganisation: graphql`
    fragment UserOrganisation_userOrganisation on CiipUserOrganisation {
      id
      status
      organisationId
      organisationByOrganisationId {
        id
        rowId
        operatorName
      }
    }
  `
});
