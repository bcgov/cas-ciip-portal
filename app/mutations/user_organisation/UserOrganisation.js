import {graphql} from 'react-relay';
import BaseMutation from '../BaseMutation';

const mutation = graphql`
  mutation UserOrganisationMutation($input: CreateUserOrganisationInput!) {
    createUserOrganisation(input: $input) {
      userOrganisation {
        id
      }
    }
  }
`;
export const userOrganisationMutation = async (environment, variables) => {
  const m = new BaseMutation(
    environment,
    mutation,
    variables,
    'create-user-organisation-mutation'
  );
  return m.performMutation();
};
