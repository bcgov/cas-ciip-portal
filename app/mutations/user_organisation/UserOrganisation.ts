import {graphql, DeclarativeMutationConfig} from 'react-relay';
import BaseMutation from '../BaseMutation';

const mutation = graphql`
  mutation UserOrganisationMutation($input: CreateCiipUserOrganisationInput!) {
    createCiipUserOrganisation(input: $input) {
      ciipUserOrganisationEdge {
        node {
          ...UserOrganisation_userOrganisation
        }
      }
    }
  }
`;

export const userOrganisationMutation = async (
  environment,
  variables,
  user
) => {
  const configs: DeclarativeMutationConfig[] = [
    {
      type: 'RANGE_ADD',
      parentID: user,
      connectionInfo: [
        {
          key: 'Organisations_ciipUserOrganisationsByCiipUserId',
          rangeBehavior: 'append'
        }
      ],
      edgeName: 'ciipUserOrganisationEdge'
    }
  ];

  const m = new BaseMutation('create-ciip-user-organisation-mutation', configs);
  return m.performMutation(environment, mutation, variables);
};
