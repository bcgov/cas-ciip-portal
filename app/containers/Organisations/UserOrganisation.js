import React from 'react';
import {graphql, createFragmentContainer} from 'react-relay';
import {Badge} from 'react-bootstrap';
import Link from 'next/link';
import {updateUserOrganisationMutation} from '../../mutations/user_organisation/UpdateUserOrganisation';

export const UserOrganisationComponent = props => {
  const {userOrganisation} = props;
  if (!userOrganisation) {
    return null;
  }

  const statusBadgeColor = {
    attention: 'warning',
    pending: 'info',
    expired: 'danger',
    active: 'success'
  };

  /** TODO: INTENDED FOR DEMO ONLY. To be removed */
  const setStatusActive = async id => {
    const {environment} = props.relay;
    const variables = {
      input: {
        id,
        userOrganisationPatch: {
          status: 'active'
        }
      }
    };
    const response = await updateUserOrganisationMutation(
      environment,
      variables
    );
    console.log(response);
  };

  if (userOrganisation.status !== 'active') {
    setTimeout(setStatusActive, 4000, userOrganisation.id);
  }
  /** End removal block */

  return (
    <tr>
      <td>{userOrganisation.organisationByOrganisationId.operatorName}</td>
      <td>
        <Badge
          pill
          style={{width: '100%', padding: '8px'}}
          variant={statusBadgeColor[userOrganisation.status]}
        >
          {userOrganisation.status}
        </Badge>
      </td>
      <td>
        {' '}
        {userOrganisation.status === 'active' ? (
          <Link
            href={{
              pathname: '/user-organisation-facilities',
              query: {
                organisationId: userOrganisation.organisationByOrganisationId.id
              }
            }}
          >
            <a className="btn btn-outline-primary" href="#">
              View Facilities
            </a>
          </Link>
        ) : null}
      </td>
      <style jsx>
        {`
          td {
            padding: 20px;
          }
        `}
      </style>
    </tr>
  );
};

export default createFragmentContainer(UserOrganisationComponent, {
  userOrganisation: graphql`
    fragment UserOrganisation_userOrganisation on UserOrganisation {
      id
      status
      organisationByOrganisationId {
        id
        operatorName
      }
    }
  `
});
