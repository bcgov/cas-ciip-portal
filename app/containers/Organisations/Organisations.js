import React from 'react';
import {graphql, createFragmentContainer} from 'react-relay';
import {Table, Dropdown, Button, Alert} from 'react-bootstrap';
import Organisation from './Organisation';
import UserOrganisation from './UserOrganisation';

export const OrganisationsComponent = props => {
  const {user, allOrganisations} = props.query;
  if (!user) return '...Loading';

  const changeInput = event => {
    event.stopPropagation();
    event.preventDefault();
    event.persist();
    props.handleInputChange(event.target.value);
  };

  const selectOrg = (name, id) => {
    props.handleInputChange(name);
    props.handleOrgChange(id);
    props.handleContextChange();
  };

  const claimOrg = async () => {
    props.handleContextChange();
    props.handleInputChange('');
    await props.handleOrgConfirm(props.relay.environment);
    props.handleOrgChange(null);
  };

  const cancelClaim = () => {
    props.handleOrgChange(null);
    props.handleContextChange();
    props.handleInputChange('');
  };

  return (
    <>
      {user.userOrganisationsByUserId.edges.length === 0 ? (
        <>
          <Alert variant="warning">
            You are not registered to apply for any Reporting Operation at this
            time. You can request access to a Reporting Operation by selecting
            one from the dropdown below.
          </Alert>
          <br />
        </>
      ) : (
        <Table
          striped
          bordered
          hover
          style={{textAlign: 'left', marginBottom: '40px'}}
        >
          <thead>
            <tr>
              <th>Operation</th>
              <th>Access Status</th>
              <th>Facilities</th>
            </tr>
          </thead>
          <tbody>
            {user.userOrganisationsByUserId.edges.map(({node}) => {
              return <UserOrganisation key={node.id} userOrganisation={node} />;
            })}
          </tbody>
          <style jsx>
            {`
              th {
                padding: 20px;
                font-weight: 500;
              }
            `}
          </style>
        </Table>
      )}

      {props.confirmOrg ? (
        <>
          <h5 className="blue">Requesting access to: </h5>
          <h4 style={{fontWeight: 300, margin: '15px 0'}}>{props.orgInput} </h4>
          <Button
            style={{marginRight: '15px'}}
            variant="primary"
            onClick={claimOrg}
          >
            Request Access
          </Button>
          <Button variant="danger" onClick={cancelClaim}>
            Cancel
          </Button>
        </>
      ) : (
        <Dropdown className="search-dropdown">
          <h5>Add an Operation:</h5>
          <Dropdown.Toggle
            as="input"
            placeholder="Search..."
            onChange={changeInput}
          ></Dropdown.Toggle>
          <Dropdown.Menu>
            {allOrganisations.edges.map(({node}) => {
              return (
                <Organisation
                  key={node.id}
                  select
                  organisation={node}
                  orgInput={props.orgInput}
                  selectOrg={selectOrg}
                />
              );
            })}
          </Dropdown.Menu>
          <style jsx global>{`
            .dropdown-toggle {
              padding: 8px 16px;
              width: 300px;
              border-radius: 3px;
              border: 1px solid #999;
            }
          `}</style>
        </Dropdown>
      )}
    </>
  );
};

export default createFragmentContainer(OrganisationsComponent, {
  query: graphql`
    fragment Organisations_query on Query
      @argumentDefinitions(id: {type: "ID!"}) {
      user(id: "WyJ1c2VycyIsMV0=") {
        id
        userOrganisationsByUserId(first: 2147483647)
          @connection(key: "Organisations_userOrganisationsByUserId") {
          edges {
            node {
              id
              ...UserOrganisation_userOrganisation
            }
          }
        }
      }

      allOrganisations {
        edges {
          node {
            id
            ...Organisation_organisation
          }
        }
      }
    }
  `
});
