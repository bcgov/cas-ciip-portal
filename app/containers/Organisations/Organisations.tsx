import React from 'react';
import {graphql, createFragmentContainer, RelayProp} from 'react-relay';
import {
  Table,
  Dropdown,
  Button,
  FormControl,
  Alert,
  Card
} from 'react-bootstrap';
import {Organisations_query} from 'Organisations_query.graphql';
import {RelayModernEnvironment} from 'relay-runtime/lib/store/RelayModernEnvironment';
import AddOrganisationFacility from 'components/AddOrganisationFacility';
import createOrganisationMutation from 'mutations/organisation/createOrganisationMutation';
import LoadingSpinner from 'components/LoadingSpinner';
import Organisation from './Organisation';
import UserOrganisation from './UserOrganisation';

interface Props {
  query: Organisations_query;
  relay: RelayProp;
  flagCertRequests: boolean;
  orgInput: string;
  selectedOrg: number;
  confirmOrg: boolean;
  handleInputChange: (event: any) => void;
  handleContextChange: () => void;
  handleOrgChange: (event: any) => void;
  handleOrgConfirm: (status: string, env: RelayModernEnvironment) => any;
}

export const OrganisationsComponent: React.FunctionComponent<Props> = (
  props
) => {
  const {session, allOrganisations} = props.query;
  if (!session) return <LoadingSpinner />;

  const changeInput = (event) => {
    event.stopPropagation();
    event.preventDefault();
    event.persist();
    props.handleInputChange(event.target.value);
  };

  const selectOrg = (name, id) => {
    props.handleInputChange(name);
    props.handleOrgChange(id);
    props.handleContextChange();
  };

  const claimOrg = async (active) => {
    props.handleContextChange();
    props.handleInputChange('');
    await props.handleOrgConfirm(active, props.relay.environment);
    props.handleOrgChange(null);
  };

  const cancelClaim = () => {
    props.handleOrgChange(null);
    props.handleContextChange();
    props.handleInputChange('');
  };

  const handleAddOrganisation = async (variables) => {
    const {environment} = props.relay;
    const response = await createOrganisationMutation(environment, variables);
    console.log(response);
    const {operatorName, rowId} = response.createOrganisation.organisation;
    selectOrg(operatorName, rowId);
  };

  const {edges} = session.ciipUserBySub.ciipUserOrganisationsByUserId;
  return (
    <>
      {props.flagCertRequests && (
        <Alert variant="info">
          One or more reporting operations has requested that you certify an
          application.{' '}
          <Alert.Link href="/certifier/requests">
            View all certification requests.
          </Alert.Link>
        </Alert>
      )}
      <Card bg="light">
        <Card.Body>
          Operator, Operator Representative, and Reporting Operation are defined
          in the{' '}
          <a href="http://www.bclaws.ca/civix/document/id/complete/statreg/14029_01">
            Greenhouse Gas Industrial Reporting and Control Act
          </a>{' '}
          and{' '}
          <a href="http://www.bclaws.ca/civix/document/id/lc/statreg/249_2015">
            Greenhouse Gas Emission Reporting Regulation
          </a>
          .
        </Card.Body>
      </Card>
      <br />
      {edges.length > 0 && (
        <Table
          striped
          bordered
          hover
          style={{textAlign: 'left', marginBottom: '40px'}}
        >
          <thead>
            <tr>
              <th>Operator</th>
              <th>Access Status</th>
              <th>Operations/Facilities</th>
            </tr>
          </thead>
          <tbody>
            {edges.map(({node}) => {
              return <UserOrganisation key={node.id} userOrganisation={node} />;
            })}
          </tbody>
          <style jsx>
            {`
              th {
                padding: 20px;
                font-weight: 500;
              }
            `}
          </style>
        </Table>
      )}

      <div>
        <Card style={{marginTop: '50px'}}>
          <Card.Body>
            {props.confirmOrg ? (
              <>
                <Card.Title>Requesting access to: </Card.Title>
                <h4 style={{fontWeight: 300, margin: '15px 0'}}>
                  {props.orgInput}{' '}
                </h4>
                {/* Dev-only button to automatically create approved user-organisation requests */}
                {process.env.NODE_ENV === 'production' ? null : (
                  <Button
                    style={{marginRight: '15px'}}
                    variant="success"
                    onClick={async () => claimOrg(true)}
                  >
                    Activate Access
                  </Button>
                )}
                <Button
                  style={{marginRight: '15px'}}
                  variant="primary"
                  onClick={async () => claimOrg(false)}
                >
                  Request Access
                </Button>
                <Button variant="danger" onClick={cancelClaim}>
                  Cancel
                </Button>
              </>
            ) : (
              <Dropdown className="search-dropdown">
                <Card.Title>
                  Request access to apply for the CIIP on behalf of a Reporting
                  Operation:
                </Card.Title>
                <Alert variant="warning">
                  <p>
                    The CleanBC Industrial Incentive Program application must be
                    submitted by the Operator of the Reporting Operation or, if
                    there is more than one Operator, the Designated Operator as
                    defined by the{' '}
                    <Alert.Link href="http://www.bclaws.ca/civix/document/id/complete/statreg/14029_01">
                      Greenhouse Gas Industrial Reporting and Control Act
                    </Alert.Link>{' '}
                    and{' '}
                    <Alert.Link href="http://www.bclaws.ca/civix/document/id/lc/statreg/249_2015">
                      Greenhouse Gas Emission Reporting Regulation
                    </Alert.Link>
                    . Other representatives of the Reporting Operation may
                    complete the application if authorization from the Operator
                    is provided.
                  </p>
                  <p>
                    Further information on the role of the Operator and
                    Certifying Official{' '}
                    <Alert.Link href="https://www2.gov.bc.ca/gov/content/environment/climate-change/industry/cleanbc-program-for-industry/cleanbc-industrial-incentive-program">
                      can be found here
                    </Alert.Link>
                    .
                  </p>
                  <p>
                    Please email{' '}
                    <Alert.Link href="mailto:GHGRegulator@gov.bc.ca?subject=CIIP Portal Inquiry">
                      GHGRegulator@gov.bc.ca
                    </Alert.Link>{' '}
                    if you have any questions.
                  </p>
                </Alert>
                <small style={{display: 'block', marginBottom: '20px'}}>
                  (You can search to narrow the results in the dropdown)
                </small>
                <Dropdown.Toggle id="org-dropdown" className="search-toggle">
                  Select Operator
                </Dropdown.Toggle>
                <Dropdown.Menu>
                  <FormControl
                    placeholder="Search..."
                    className="mx-3 my-2 w-auto"
                    onChange={changeInput}
                  />
                  <div className="org-scroll">
                    {allOrganisations.edges.map(({node}) => {
                      return (
                        <Organisation
                          key={node.id}
                          select
                          organisation={node}
                          orgInput={props.orgInput}
                          selectOrg={selectOrg}
                        />
                      );
                    })}
                  </div>
                </Dropdown.Menu>
              </Dropdown>
            )}
            <hr />
            <AddOrganisationFacility
              onAddOrganisation={handleAddOrganisation}
            />

            <style jsx>
              {`
                .org-scroll {
                  max-height: 250px;
                  overflow: hidden;
                  overflow-y: scroll;
                }
              `}
            </style>
          </Card.Body>
        </Card>
      </div>
    </>
  );
};

export default createFragmentContainer(OrganisationsComponent, {
  query: graphql`
    fragment Organisations_query on Query {
      session {
        ciipUserBySub {
          id
          ciipUserOrganisationsByUserId(first: 2147483647)
            @connection(key: "Organisations_ciipUserOrganisationsByUserId") {
            edges {
              node {
                id
                ...UserOrganisation_userOrganisation
              }
            }
          }
        }
      }

      allOrganisations {
        edges {
          node {
            id
            ...Organisation_organisation
          }
        }
      }
    }
  `
});
