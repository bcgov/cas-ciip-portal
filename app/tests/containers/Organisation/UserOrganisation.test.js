import React from 'react';
import {shallow} from 'enzyme';
import {UserOrganisationComponent} from '../../../containers/Organisations/UserOrganisation';

describe('UserOrganisation', () => {
  it('should render null if no userOrganisation is passed as props', async () => {
    const userOrganisation = null;
    const r = shallow(
      <UserOrganisationComponent userOrganisation={userOrganisation} />
    );
    expect(r).toMatchSnapshot();
    expect(r.exists('Badge')).toEqual(false);
  });
  it('should render a userOrganisation if the userOrganisation is passed as props', async () => {
    const userOrganisation = {
      status: 'active',
      organisationByOrganisationId: {
        id: 'orgId'
      }
    };
    const r = shallow(
      <UserOrganisationComponent userOrganisation={userOrganisation} />
    );
    expect(r).toMatchSnapshot();
    expect(r.find('Badge').text()).toBe('active');
  });
});
