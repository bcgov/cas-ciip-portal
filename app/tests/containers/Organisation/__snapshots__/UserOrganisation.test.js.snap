// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserOrganisation should render a userOrganisation if the userOrganisation is passed as props 1`] = `
<tr
  className="jsx-2428472905"
>
  <td
    className="jsx-2428472905"
  />
  <td
    className="jsx-2428472905"
  >
    <Badge
      pill={true}
      style={
        Object {
          "padding": "8px",
          "width": "100%",
        }
      }
      variant="success"
    >
      active
    </Badge>
  </td>
  <td
    className="jsx-2428472905"
  >
     
    <Link
      href={
        Object {
          "pathname": "/user-organisation-facilities",
          "query": Object {
            "organisationId": "orgId",
          },
        }
      }
    >
      <a
        className="jsx-2428472905 btn btn-outline-primary"
        href="#"
      >
        View Facilities
      </a>
    </Link>
  </td>
  <JSXStyle
    id="2428472905"
  >
    td.jsx-2428472905{padding:20px;}
  </JSXStyle>
</tr>
`;

exports[`UserOrganisation should render null if no userOrganisation is passed as props 1`] = `""`;
